azurerm/internal/services/web/app_service_resource_test.go

```diff
@@ -1862,6 +1862,39 @@ func TestAccAppService_basicWindowsContainer(t *testing.T) {
 	})
 }
 
+func TestAccAppService_AcrManageIdentityCredentials(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_service", "test")
+	r := AppServiceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.AcrUseManagedIdentity(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
+				check.That(data.ResourceName).Key("identity.0.principal_id").MatchesRegex(validate.UUIDRegExp),
+				check.That(data.ResourceName).Key("identity.0.tenant_id").MatchesRegex(validate.UUIDRegExp),
+				check.That(data.ResourceName).Key("site_config.0.acr_use_managed_identity_creds").HasValue("true"),
+			),
+		},
+	})
```

Comment:
We can make use of `ImportStep` here to check values:
```suggestion
	data.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.AcrUseManagedIdentity(data),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
		},
		data.ImportStep(),
	})
```

Reply:
I am new to Go. Can you explain how importSteps will validate the property values return?

Reply:
It's a part of our test framework, rather than a Go specific thing. The `ImportStep` helper function reads the resource back and if there are differences between that view and what is configured, it errors out with the differences. It optionally takes a list of schema keys that are not returned by the API and should be ignored for the comparison. (e.g. if a read on a resource doesn't return a sensitive value like a password). Does that help?

Reply:
That answered my question.